### ISSUE TITLE
NIPS 25 | Scaling RL to Long Videos

### ISSUE THREAD

**Issue by hiahei** [AI-Infra-Team/awesome-papers#136]
# Scaling RL to Long Videos
Authors: 
Yukang Chen, Wei Huang, Baifeng Shi, Qinghao Hu, Hanrong Ye, Ligeng Zhu, Zhijian Liu, Pavlo Molchanov, Jan Kautz, Xiaojuan Qi, Sifei Liu, Hongxu Yin, Yao Lu, Song Han
[Paper](https://arxiv.org/abs/2507.07966)

### 1 Motivation
Reasoning is indispensable for long video understanding that goes beyond recognition alone.

RL for long videos presents significant challenges due to the high computational demands of processing hundreds to thousands of frames. Existing RL frameworks struggle with such longcontext training in rollout and LLM prefilling.

### 2 LongVideo-Reason Data
- Q&A pairs are categorized into four reasoning types: Temporal, Goal and Purpose, Spatial, and Plot and Narrative Reasoning. An LLM is then used to refine and streamline the reasoning steps.
 
<div align="center">
<img width="1129" height="214" alt="Image" src="https://github.com/user-attachments/assets/7bffdbcf-2c93-49c5-bf04-de6deb955856" />
</div>

- To avoid gradient vanishing for GRPO, we choose data with diverse responses.

### 3 LongVILA Training Pipeline
LongVILA-R1: 
1) warm-up for long video reasoning, utilizing 36K data with high quality CoT for SFT on the MM-SP system;
2) reinforcement learning with dense frames from long videos.

<div align="center">
<img width="1141" height="215" alt="Image" src="https://github.com/user-attachments/assets/5b7a8593-100d-4a97-9e26-a67013978b15" />
</div>

We perform SFT on hundreds of frames adopting MM-SP training system from LongVILA, serving as a warm-up phase for subsequent RL. This stage equips the model with fundamental reasoning abilities and instructionfollowing skills for long video scenarios.

### 4 Multi-modal Reinforcement SP
Long videos present unique challenges due to their high token volume. We develop the MR-SP framework that scales reinforcement learning for long-context video reasoning.

MR-SP leverages sequence parallelism in both rollout and pre-filling stages, enabling long videos in RL, with reduced overhead.

#### Stage 1 - Rollout with Paralleled Encoding
We adopt sequence parallelism (SP) for the video encoding stage.

The input video frames are first evenly divided across multiple GPUs (e.g., GPU 1 to GPU 3), each equipped with its own vision tower. Each GPU independently processes a slice of the video, encoding only a subset of the frames.

After the video embeddings are globally gathered, they are reused for downstream usage throughout the RL pipeline.

<div align="center">
<img width="1131" height="393" alt="Image" src="https://github.com/user-attachments/assets/771f1242-4c11-444f-b59a-586b7629e667" />
</div>

#### Stage 2 - Prefilling with Sequence Parallelism
For each rollout, both the reference and policy models require prefilling compute-intensively in RL for long videos. With the gathered embedding from Stage 1 reused, we parallelize the inference stage across devices using sequence parallelism. As illustrated in below figure, we globally gathered input embeddings are first padded to a uniform length (Padding Sequence) and then evenly partitioned across GPUs (Sharding to Local GPU). This allows each GPU to handle only a portion of the input sequence during prefilling. This parallelism is applied to both policy and reference model prefilling. Then, each GPU locally computes logits for its token slice, prefilling in parallel.

### 5 Experiment

<div align="center">
<img width="1393" height="352" alt="Image" src="https://github.com/user-attachments/assets/7b33c787-e4e4-476a-bacd-88353d720b96" />
</div>

<div align="center">
<img width="1134" height="265" alt="Image" src="https://github.com/user-attachments/assets/e46747ac-4d9d-4d69-a86f-988bf0becf7d" />
</div>

Using only Stage 1 of MR-SP significantly improves efficiency up to 512 frames but encounters GPU out-of-memory (OOM) issues beyond that point. In contrast, the full MR-SP system consistently reduces runtime, achieving up to a 2.1× speedup at 512 frames and scaling efficiently to 1024 frames without OOM, highlighting the benefit of combining sequence reuse and sequence parallelism for long video RL training.

<div align="center">
<img width="1407" height="521" alt="Image" src="https://github.com/user-attachments/assets/c3c1ef01-cb91-4458-87df-808d927bcb81" />
</div>
